[docs] Add docs for Upgrade Assistant APIs (#30330)
elastic/kibana: docs/api/upgrade-assistant/status.asciidoc

[[upgrade-assistant-api-status]]
=== Upgrade Readiness Status

> experimental[This API is *experimental* and may be changed or removed completely in a future release. The underlying Spaces concepts are stable, but the APIs for managing Spaces are currently experimental.]

Reviewer: ```suggestion
experimental[This API is *experimental* and may be changed or removed completely in a future release. The underlying Upgrade Assistant concepts are stable, but the APIs for managing Upgrade Assistant are currently experimental.]
```

==== Request

To check the status of your cluster, submit a GET request to the `/api/upgrade_assistant/status` endpoint:

Note: You cannot access this endpoint via the Console in Kibana.

[source,js]
--------------------------------------------------
GET /api/upgrade_assistant/status
--------------------------------------------------
// KIBANA

==== Response

A successful call returns a response code of `200` and a response body
containing a JSON structure similar to the following example:

[source,js]
--------------------------------------------------
{
  "readyForUpgrade": false,
  "cluster": [
    {
      "message": "Cluster deprecated issue",
      "details": "...",
      "level": "warning",
      "url": "https://docs.elastic.co/..."
    }
  ],
  "indices": [
    {
      "message": "Index was created before 6.0",
      "details": "...",
      "index": "myIndex",
      "level": "critical",
      "reindex": true, <1>
      "url": "https://docs.elastic.co/..."
    }
  ]
}
--------------------------------------------------

> <1> Indices with the `reindex` attribute set to `true` should be fixed using the <<upgrade-assistant-api-reindexing>>.

Reviewer: ```suggestion
<1> You can fix indices with the `reindex` attribute set to `true` using the <<upgrade-assistant-api-reindexing>>.
```
